internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -655,6 +655,42 @@ func getHyperthreading(clusterInstall *hiveext.AgentClusterInstall) *string {
 	}
 }
 
+func (r *ClusterDeploymentsReconciler) getEncodedCACert(ctx context.Context,
+	log logrus.FieldLogger,
+	caCertificateRef *hiveext.CaCertificateReference) (*string, error) {
+	caSecret, err := getSecret(ctx, r.Client, r.APIReader, types.NamespacedName{Namespace: caCertificateRef.Namespace, Name: caCertificateRef.Name})
+	if err != nil {
+		return nil, errors.Wrap(err, "error getting ca certificate secret")
+	}
+	caCertBytes, hasCACert := caSecret.Data[corev1.TLSCertKey]
+	if !hasCACert {
+		return nil, fmt.Errorf("CA Secret is missing tls.crt key")
+	}
+	encodedCACert := base64.StdEncoding.EncodeToString(caCertBytes)
+	caCertificate := swag.String(encodedCACert)
+	return caCertificate, nil
+}
+
+func (r *ClusterDeploymentsReconciler) getIgnitionEndpointFromKubeapi(ctx context.Context,
+	log logrus.FieldLogger,
+	kubeapiIgnitionEndpoint *hiveext.IgnitionEndpoint) (*models.IgnitionEndpoint, error) {
+
+	ignitionEndpoint := &models.IgnitionEndpoint{}
+	if kubeapiIgnitionEndpoint.Url != "" {
```

Comment:
Similar to above, allow unsetting the URL (if it's nil and the clusterInstall.Spec.IgnitionEndpoint.Url isn't nil)

Reply:
Done